File: Source/Extensions/Tessa.Extensions.Default.Client/Tiles/KrEditModeTileExtension.cs
using System.Threading.Tasks;
using Tessa.Extensions.Default.Shared;
using Tessa.Extensions.Default.Shared.Workflow.KrPermissions;
using Tessa.UI;
using Tessa.UI.Tiles;
using Tessa.UI.Tiles.Extensions;

namespace Tessa.Extensions.Default.Client.Tiles
{
    public sealed class KrEditModeTileExtension :
        TileExtension
    {
        #region Command Actions

        private static async void OpenForEditingAsync(object parameter)
        {
            await KrTileHelper.OpenMarkedCardAsync(KrPermissionsHelper.CalculatePermissionsMark,
                null, //Не требуем подтверждения действия, если не было изменений
                () => TessaDialog.ConfirmWithCancel("$KrTiles_EditModeConfirmation"));
        }

        #endregion

        #region Base Overrides

        public override Task InitializingGlobal(ITileGlobalExtensionContext context)
        {
            ITileContextSource contextSource = context.Workspace.LeftPanel;

            context.Workspace.LeftPanel.Tiles.Add(
                new Tile(
                    DefaultTileNames.KrEditMode,
                    TileHelper.SplitCaption("$KrTiles_EditMode"),
                    context.Icons.Get("Thin2"),
                    contextSource,
                    new DelegateCommand(OpenForEditingAsync),
                    TileGroups.Cards,
                    order: 22,
                    toolTip: TileHelper.GetToolTip("$KrTiles_EditModeTooltip", KrTileKeys.Edit),
                    evaluating: TileHelper.EnableWhenVisibleInCardHandler));

            return Task.CompletedTask;
        }


        public override Task InitializingLocal(ITileLocalExtensionContext context)
        {
            ITile krEnterEditMode = context.Workspace.LeftPanel.Tiles.TryGet(DefaultTileNames.KrEditMode);
            if (krEnterEditMode != null)
            {
                krEnterEditMode.Context.AddInputBinding(krEnterEditMode, KrTileKeys.Edit);
            }

            return Task.CompletedTask;
        }

        #endregion
    }
}

File: WebClient SDK/src/default/tiles/krEditModeTileExtension.ts
import { showConfirmWithCancel } from 'tessa/ui';
import { TileExtension, ITileGlobalExtensionContext, Tile, TileGroups,
  ITileLocalExtensionContext, TileHotkey, openMarkedCard, enableWhenVisibleInCardHandler } from 'tessa/ui/tiles';

export class KrEditModeTileExtension extends TileExtension {

  //#region TileExtension

  public initializingGlobal(context: ITileGlobalExtensionContext) {
    const contextSource = context.workspace.leftPanel.contextSource;

    context.workspace.leftPanel.tiles.push(
      new Tile({
        name: 'KrEditMode',
        caption: '$KrTiles_EditMode',
        icon: 'ta icon-thin-002',
        contextSource,
        command: KrEditModeTileExtension.openForEditing,
        group: TileGroups.Cards,
        order: 22,
        evaluating: enableWhenVisibleInCardHandler,
        toolTip: '$KrTiles_EditModeTooltip'
      })
    );
  }

  public initializingLocal(context: ITileLocalExtensionContext) {
    const panel = context.workspace.leftPanel;
    const hotkeyStorage = panel.contextSource.hotkeyStorage;
    const krEnterEditMode = panel.tryGetTile('KrEditMode');
    if (krEnterEditMode) {
      hotkeyStorage.addTileHotkey(new TileHotkey(krEnterEditMode, 'Alt+E', 'KeyE', {alt: true}));
    }
  }

  //#endregion

  //#region actions

  private static async openForEditing() {
    await openMarkedCard(
      'kr_calculate_permissions',
      null, // Не требуем подтверждения действия, если не было изменений
      () => showConfirmWithCancel('$KrTiles_EditModeConfirmation')
    );
  }

  //#endregion

}